framework/src/meshgenerators/AdvancedExtruderGenerator.C

```diff
@@ -621,6 +748,26 @@ FancyExtruderGenerator::generate()
         // maintain the subdomain_id
         new_elem->subdomain_id() = elem->subdomain_id();
 
+        // define upward boundaries
+        if (k == num_layers - 1)
+        {
+          const unsigned short top_id =
```

Comment:
top_id is the id of the top side right?
pick a more explicit name or add a comment